### ISSUE TITLE
use exec id instead of pid and wait on SIGUSR1

### ISSUE THREAD

**Issue by bergwolf** [kata-containers/shim#18]
As we discussed in kata-containers/agent#72, we will use runtime generated exec id to uniquely identify a process, replacing current pid returned by agent.

And in order to support `runtime create --pid-file` option, we need to start shim before starting the container. So pause shim on SIGUSR1 if it is an init process shim, in which case container ID equals to exec ID. Runtime should wait up shim after it calls `StartContainer()`.

**Comment by sameo** [comment]
So here we're putting 2 constraints on the runtime:

1. It must send a SIGUSR1 to the shim pid at some precise point in the code
2. It must set exec ID == container ID in some specific case (init process)

and IIUC we do that so that we pause the shim before it waits for a potentially non existing process and before it starts reading output streams from this same non existing process.
Please correct me if I am misunderstanding the issue.

I think this should all be handled by the agent itself and both the shim and the runtime should not have special code path for waiting for some obscure signal that a runtime may send at some point. The agent should be able to put a `WaitProcess()` caller on hold until the actual process shows up. It would add a little complexity on the agent side but would simplify and generalize the shim and runtime code.

cc @sboeuf
